MapboxCoreNavigation/RouteController.swift

```diff
@@ -330,6 +332,54 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func checkForFasterRoute(from location: CLLocation) {
+        
+        // If the user is on a short route, don't check for faster alternatives
+        guard routeProgress.durationRemaining > 600 else { return }
+
+        // If the user is approaching a maneuver, don't check for a faster alternatives
+        guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining > 70 else { return }
+        
+        guard let currentUpcomingManeuver = routeProgress.currentLegProgress.upComingStep else { return }
+        
+        guard let lastLocationTimestamp = lastLocationTimestamp else {
+            self.lastLocationTimestamp = location.timestamp
+            return
+        }
+
+        // Only check ever 2 minutes for faster route
+        guard location.timestamp.timeIntervalSince1970 - lastLocationTimestamp.timeIntervalSince1970 > 120 else { return }
+        
+        let options = routeProgress.route.routeOptions
+        options.waypoints = [Waypoint(coordinate: location.coordinate)] + routeProgress.remainingWaypoints
+        if let firstWaypoint = options.waypoints.first, location.course >= 0 {
```

Comment:
`firstWaypoint` is guaranteed to be the one we created above using `Waypoint(coordinate:)`. Pull it out into a local variable and reuse it below.